File: 693.binary-number-with-alternating-bits.go
package main

/*
 * @lc app=leetcode id=693 lang=golang
 *
 * [693] Binary Number with Alternating Bits
 */

// @lc code=start
func hasAlternatingBits(n int) bool {
	prevBit := n & 1
	n = n >> 1
	for n > 0 {
		currBit := n & 1
		if currBit == prevBit {
			return false
		}
		prevBit = currBit
		n = n >> 1
	}
	return true
}

// @lc code=end

File: 693.binary-number-with-alternating-bits.rb
#
# @lc app=leetcode id=693 lang=ruby
#
# [693] Binary Number with Alternating Bits
#

# @lc code=start
# @param {Integer} n
# @return {Boolean}
def has_alternating_bits(n)
  memo = 0
  n.to_s(2).each_char do |bit|
    return false if bit == memo 

    memo = bit
  end
  return true
end
# @lc code=end

